pkg/wwan/mmagent/mmdbus/client.go

```diff
@@ -749,8 +749,14 @@ func (c *Client) getModemPhysAddrs(modemObj dbus.BusObject) (
 	devPathSymlink := filepath.Join("/sys/class/usbmisc", primaryPort, "device")
 	devPath, err := filepath.EvalSymlinks(devPathSymlink)
 	if err != nil {
-		return addrs, proto, fmt.Errorf("failed to eval symlink %s: %w",
-			devPathSymlink, err)
+		// Maybe this modem uses PCIe interface as opposed to USB.
+		// This is more and more common for modern high-speed 4G and 5G modems.
+		devPathSymlink = filepath.Join("/sys/class/wwan", primaryPort, "device")
+		devPath, err = filepath.EvalSymlinks(devPathSymlink)
+		if err != nil {
+			return addrs, proto, fmt.Errorf("failed to eval symlink %s: %w",
```

Comment:
This just prints the PCIe path. Would it make sense to include both paths which were tried and failed in the error string?

Reply:
Makes sense, done.